Add workshop setup instructions to the website (#1)
EDUCE-UBC/EDUCE-UBC.github.io: reproducibility_ws_setup.Rmd

---
title: "Workshop setup instructions"
subtitle: "Workshop: Reproducible research"
output: 
  html_document:
    toc: yes
params:
  r_version: 4.0.2
  r_legacy_version: 3.3.3
  rstudio_version: 1.3.1073

>   rstudio_legacy_version: 1.1.463

Reviewer: ```suggestion
params:
  make_version: 4.3
```
`params` for R and RStudio are not needed. Use params for the latest version of `make` and use it below.

---

## Instructions for all users

1. Make sure that you have installed the [latest versions of R and RStudio](r_and_rstudio.html). 
2. Install the tidyverse and the renv packages by copying the following code into the Console, after the prompt `>`

  ```{r install-tidyverse, eval = FALSE}
  if (!require(tidyverse)) install.packages("tidyverse")
  if (!require(renv)) install.packages("renv")
  ```

3. Download the data file we will use during the workshop by clicking on the following button and save it in your Desktop or another convenient location.
  
```{r echo = FALSE}
data_file <- read.csv("https://raw.githubusercontent.com/EDUCE-UBC/workshop_data/master/data.csv")

downloadthis::download_this(data_file,
    output_name = "data",
    output_extension = ".csv",
    button_label = "Download data",
    button_type = "default",
    has_icon = TRUE,
    icon = "fa fa-save",
    class = "hvr-sweep-to-left",
    csv2 = FALSE
  )
```

## Instructions for Windows users

### Terminal

> Install [GitBash](https://gitforwindows.org/), which emulates Bash (the Unix shell) in Windows. All command line instructions should be completed in GitBash and not the native Windows Command Prompt.

Reviewer: ```suggestion
Install [GitBash](https://gitforwindows.org/), which emulates Bash (a Unix shell) in Windows. All command-line instructions should be completed in GitBash and not the native Windows Command Prompt.
```

2. Test that Git is properly installed by opening GitBash and typing `git --version` [Enter]. This should populate the window with information on your version of Git such as

```
git version 2.20.1.windows.1
```

3. Once you've installed Git, you'll need to activate it in RStudio:
    - Open RStudio and go to Tools > Global Options > Git/SVN
    - Click "Enable version control interface for RStudio projects"
    - If necessary, enter the path for your Git executable where provided.
        - The path should look approximately like `C:/Program Files/Git/bin/git.exe`
        - Warning: do **not** use `C:/Program Files/Git/cmd/git.exe`. The path should include `bin`, **not** `cmd`.
        - Warning: the path should end in `git.exe`, **not** in `git-bash.exe`
    - Restart RStudio, if it worked out you will find the Git icon on the top toolbar, as shown below.

![](images/setup_git_windows.png)

### GitHub

1. If you do not already have one, sign up for an account on [GitHub](https://github.com/).
2. Configure your GitHub user name and email (the email address you used to register on GitHub). To do this, open GitBash and type:

```
git config --global user.name 'yourGitHubUsername'

> git config --global user.email 'name@provider.com'

Reviewer: ```suggestion
git config --global user.name '<your GitHub username>'
git config --global user.email '<name@provider.com>'
```

```

### Make

1. Download the [latest version of Make](https://sourceforge.net/projects/ezwinports/files/) *without guile*.

>   * The file name looks like `make-4.1-2-without-guile-w32-bin.zip`, though the most recent version number may be different.

Reviewer: ```suggestion
1. Download the [Make *without guile* (version `r params$make_version)`](https://sourceforge.net/projects/ezwinports/files/), file name `` `r paste0("make-", params$make-version, "-without-guile-w32-bin.zip")` ``
```
Here an example of how to use a params variable.

2. Unzip the folder.

> 3. Copy all of its contents to `c/Program Files/Git/mingw64/`

Reviewer: ```suggestion
3. Copy all of its contents to `C:\Program Files\Git\mingw64\`.
```

I think it is better to stick with Windows formatting of file paths.

  * Merge all of the contents of the folders but do NOT overwrite/replace any existing files.
4. Check that Make is installed by typing `make --version` [Enter] in GitBash. This should populate the window with information on your version of Make such as

```

> GNU Make 4.2.1

Reviewer: ```suggestion
GNU Make 4.3
```

Author: Is there any way to use `params$make_version` within a code chunk delimited by ```, such as this one?

Reviewer: I believe you can just make another chunk with either one or two backticks `` ` ``.

Reviewer: e.g. `` `r params$make_version` ``.

Author: Thanks! I fixed it now, and also fixed another `params`-related error that was stopping the file from building (it builds fine now).

Built for Windows32
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL verison 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

## Instructions for Mac users

### Terminal

> The native Terminal in the Mac OS is Linux-based and therefore, can be used in this workshop. You can access it under Applications > Utilities > Terminal.app

Reviewer: ```suggestion
The native Terminal on macOS is Unix-based and, therefore, can be used in this workshop. You can access it under Applications > Utilities > Terminal.app
```

### Git

1. Download and [install Git](https://git-scm.com/downloads). 
2. Test that Git is properly installed by opening your Terminal and typing `git --version` [Enter]. This should populate the window with information on your version of Git such as

```
git version 2.18.0
```

3. Once you've installed Git, you'll need to activate it in RStudio:
  * Open RStudio and go to Tools > Global Options > Git/SVN
  * Click "Enable version control interface for RStudio projects"
  * If necessary, enter the path for your Git executable where provided (see picture below).

>   * Restart RStudio, if it worked out you will find the Git icon on the top toolbar, as shown below.

Reviewer: Use `-` for bullet points.

![](images/setup_git_mac.png)

### GitHub

1. If you do not already have one, sign up for an account on [GitHub](https://github.com/).
2. Configure your GitHub user name and email (the email address you used to register on GitHub). To do this, open the Terminal and type:

```
git config --global user.name 'yourGitHubUsername'

> git config --global user.email 'name@provider.com'

Reviewer: ```suggestion
git config --global user.name '<your GitHub username>'
git config --global user.email '<name@provider.com>'
```

```

### Make

> 1. Download and install the Mac command line developer tools (including Make) by opening your Terminal and typing: `xcode-select --install` [Enter]. Click "Install" to allow the installation of the full developer’s packet.

Reviewer: You should clarify that a window opens (outside of the terminal).
